### PR TITLE
linter: migrate tslint to eslint

### PR DESCRIPTION
**Pull request by holiber**
I disabled some new ESlint's rules that affects many files to be invalid.
If some rule affects not a big amount of files than I fixed them or added `eslint-disable` directive

### CODE REVIEW

File: app/components/shared/Hotkey.vue.ts [hunk 1]
```diff
@@ -45,14 +45,15 @@ export default class HotkeyComponent extends TsxComponent<HotkeyProps> {
...
+      /*eslint-disable*/
       event instanceof MouseEvent
         ? {
             1: 'MiddleMouseButton',
             3: 'X1MouseButton',
             4: 'X2MouseButton',
           }[event.button]
         : event.code;
-
+      /*eslint-enable*/
```
Comment (andycreeth) [inline app/components/shared/Hotkey.vue.ts]: Will the `eslint-disable` run until the end of the file if this is accidentally removed?  Seems a little dangerous, but hopefully eslint throws an error if you omit the closing tag?

File: app/components/windows/EditStreamInfo.vue.ts [hunk 2]
```diff
@@ -144,10 +144,12 @@ export default class EditStreamInfo extends Vue {
         required: true,
         disabled: this.updatingInfo,
         description: this.isFacebook
+          /* eslint-disable */
```
Comment (andycreeth) [inline app/components/windows/EditStreamInfo.vue.ts]: What doesn't eslint like about this?  Can't we just format it the way it wants?

File: app/services/api/jsonrpc/jsonrpc.ts [hunk 3]
```diff
@@ -107,6 +116,7 @@ export class JsonrpcService extends Service implements IJsonrpcServiceApi {
     data: any;
     isRejected?: boolean;
   }): IJsonRpcResponse<IJsonRpcEvent> {
+    // eslint-disable-next-line
     return JsonrpcService.createResponse.apply(this, arguments);
```
Comment (andycreeth) [inline app/services/api/jsonrpc/jsonrpc.ts]: Curious why we need to disable so much in this file.

File: .eslintrc.js [hunk 4]
```diff
@@ -0,0 +1,115 @@
...
+    // We do imports where our files are suffixed .vue and this rule would expect we import as e.g. MainVue
+    "import-name": OFF,
+    // TODO: fix instances instead
+    "no-increment-decrement": OFF,
+    // This is no longer required on refactor/mutation-names branch TODO: remove
+    "function-name": OFF,
+    "no-boolean-literal-compare": OFF,
+
+    // these rules have been disable after the migration from tslint
+    // TODO: make decision about what rules we should enable
```
[the remark this answers is not in the document]
Comment (holiber) [reply .eslintrc.js]: yes. If you enable one of these rules you will see many errors in the codebase

File: .eslintrc.js [hunk 5]
```diff
@@ -0,0 +1,115 @@
...
+    parser: '@typescript-eslint/parser',
+    sourceType: 'module',
+  },
+  plugins: [
+    'vue',
+    '@typescript-eslint',
+    'prettier'
+  ],
+  rules: {
+    // TODO: enable prettier when the prettier plugin will support optional chaning
```
[the remark this answers is not in the document]
Comment (holiber) [reply .eslintrc.js]: should be fixed

File: app/components/windows/EditStreamInfo.vue.ts (hunk 2 above)
Comment (holiber) [reply app/components/windows/EditStreamInfo.vue.ts]: `ESlint` requires 12 spaces of indentation but `Eslint:prettier` plugin requires 14 spaces here.
That looks like conflicting rules that I can't fix in the code. This needs more investigation in the prettier config, but it seems like it's only one place in the code where we have this conflict.

File: app/services/api/jsonrpc/jsonrpc.ts (hunk 3 above)
Comment (holiber) [reply app/services/api/jsonrpc/jsonrpc.ts]: Using of `arguments` is not allowed. Each disabled line contains the `arguments` keyword

File: app/components/shared/Hotkey.vue.ts (hunk 1 above)
Comment (holiber) [reply app/components/shared/Hotkey.vue.ts]: Unfortunately, it doesn't throw an error

**Pull request merged**